class15/jihoi-kang/aas/app/src/main/java/com/jay/aas/ui/movie/MovieViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.jay.aas.ui.movie
+
+import androidx.databinding.ObservableField
+import androidx.lifecycle.viewModelScope
+import com.jay.aas.base.BaseViewModel
+import com.jay.aas.data.MovieRepository
+import com.jay.aas.model.Movie
+import kotlinx.coroutines.launch
+
+class MovieViewModel(
+    private val movieRepository: MovieRepository
+) : BaseViewModel() {
+
+    val movieItems = ObservableField<List<Movie>>(emptyList())
+    val movieDetailLink = ObservableField<String>()
+    val query = ObservableField<String>()
+
+    val hideKeyboardEvent = ObservableField<Unit>()
+    val showSearchFailedEvent = ObservableField<Unit>()
+    val startSearchHistoryEvent = ObservableField<Unit>()
+
+    fun getMovies() {
+        viewModelScope.launch {
+            val movies = movieRepository.getMovies()
+            movieItems.set(movies)
+        }
+    }
+
+    fun searchMovies() {
+        query.get() ?: return
+        if (query.get()!!.isEmpty()) return
```

Comment:
val 로 선언해서 사용한다면 !!는 없어도 됩니다

Reply:
Smart cast to 'String' is impossible, because 'query.get()' is a complex expression
위와 같은 error와 함께 !! 사용하라고 뜹니다. 위는 `val query = ObservableField<String>()` 요렇게 정의되어 있습니다.

Reply:
val movieQuery = query.get() ?: return
하면 안되나요?

Reply:
08cbdd3
아하, 수정하였습니다. 감사합니다.